### PR TITLE
Adds 4e algorithms.

### PR DESCRIPTION
**Pull request by samagra14**
Adds algorithms from 4th edition chapters, numbered 3 and 4.

### CODE REVIEW

File: md/Depth-Limited-Search.md [hunk 1]
```diff
@@ -1,5 +1,28 @@
...
+__function__ DEPTH-LIMITED-SEARCH(_problem_, _l_) __returns__ a solution, or failure, or cutoff  
+&emsp;_frontier_ &larr; a FIFO queue initially containing one path, for the _problem_'s initial state  
+&emsp;_solution_ &larr; failure  
+&emsp;__while__  _frontier_ is not empty __do__  
+&emsp;&emsp;&emsp;_parent_ &larr; pop(_frontier_)  
+&emsp;&emsp;&emsp;__if__ depth(_parent_) > l __then__  
+&emsp;&emsp;&emsp;&emsp;&emsp;_solution_ &larr; cutoff  
+&emsp;&emsp;&emsp;&emsp;&emsp;__else__  
+&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;__for__ _child_ __in__ successors(_parent_) __do__  
+&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;__if__ _child_ is a goal __then__
```
Comment (samagra14) [inline md/Depth-Limited-Search.md]: @norvig 
In the rest of the chapter, we check if a state is a goal state or not. However, in this algorithm, we have applied the test directly on a node. The test is applied as follows in the rest of the algorithms:
&emsp;__for__ _child_ __in__ successors(_parent_) __do__  
&emsp;&emsp;&emsp; _s_ &larr; _child_.state  
&emsp;&emsp;&emsp;__if__ _s_ is a goal  __then__  
and then the rest of the code. Please have a look into this.

Comment (norvig) [reply md/Depth-Limited-Search.md]: I think it should be child.state here too.

**Comment by norvig** [comment]
Looks good!

File: md/Breadth-First-Search.md [hunk 2]
```diff
@@ -1,5 +1,27 @@
...
+## AIMA4e
+
+__function__ BREADTH-FIRST-SEARCH(_problem_) __returns__ a solution, or failure  
+&emsp;__if__ problem's initial state is a goal __then return__ empty path to initial state  
+&emsp;_frontier_ &larr; a FIFO queue initially containing one path, for the _problem_'s initial state  
+&emsp;_reached_ &larr; a set of states; initially empty  
+&emsp;_solution_ &larr; failure  
+&emsp;__while__  _frontier_ is not empty __do__  
+&emsp;&emsp;&emsp;_parent_ &larr; the first node in _frontier_  
+&emsp;&emsp;&emsp;__for__ _child_ __in__ successors(_parent_) __do__
```
[the remark this answers is not in the document]
Comment (norvig) [reply md/Breadth-First-Search.md]: Yes, we should probably be consistent, and thorough, and use problem as an argument to successors everywhere.

**Pull request merged**